### PR TITLE
Cluster resize on start-up was fixed for all clusters

### PR DESCRIPTION
**Pull request by RostislavPorohnya**
Also made some minor fixes for some cluster resources

### CODE REVIEW

File: apis/clusters/v1beta1/opensearch_types.go [hunk 1]
```diff
@@ -283,6 +283,10 @@ func (oss *OpenSearchSpec) areDCsEqual(b []*OpenSearchDataCentre) bool {
 	}
 
 	for i := range b {
+		// TODO remove nodes number and node size from OpenSearch data centre
+		b[i].DataCentre.NodesNumber = a[i].DataCentre.NodesNumber
+		b[i].DataCentre.NodeSize = a[i].DataCentre.NodeSize
```
Comment (testisnullus) [inline apis/clusters/v1beta1/opensearch_types.go:288]: Do you want to compare these fields or what?

File: controllers/clusters/cadence_controller.go [hunk 2]
```diff
@@ -60,10 +60,6 @@ type CadenceReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the Cadence object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
```
Comment (testisnullus) [inline controllers/clusters/cadence_controller.go:66]: Why did you remove this comment?

**Review by testisnullus — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/opensearch_types.go (hunk 1 above)
Comment (RostislavPorohnya) [reply apis/clusters/v1beta1/opensearch_types.go]: This fields should be removed for opensearch. I need them to be equal to pass check

Comment (RostislavPorohnya) [reply apis/clusters/v1beta1/opensearch_types.go]: Removed this check since fields are removed

File: controllers/clusters/cadence_controller.go (hunk 2 above)
Comment (RostislavPorohnya) [reply controllers/clusters/cadence_controller.go:66]: I think It's unneeded part of comment, we have removed it in other controllers

File: controllers/clusters/postgresql_controller.go [hunk 3]
```diff
@@ -286,6 +286,10 @@ func (r *PostgreSQLReconciler) handleCreateCluster(
 		return models.ReconcileRequeue
 	}
 
+	if pg.Annotations[models.ResourceStateAnnotation] == models.UpdatingEvent {
+		return models.ReconcileRequeueNow
+	}
```
Comment (ribaraka) [inline controllers/clusters/postgresql_controller.go:291]: why do you need to go straight to update?

File: config/samples/clusters_v1beta1_cassandra.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@ kind: Cassandra
 metadata:
   name: cassandra-cluster
 spec:
-  name: "Username-Cassandra"
+  name: "rostyslp-Cassandra"
```
[the remark this answers is not in the document]
Comment (RostislavPorohnya) [reply config/samples/clusters_v1beta1_cassandra.yaml:6]: Ok

File: controllers/clusters/postgresql_controller.go (hunk 3 above)
Comment (RostislavPorohnya) [reply controllers/clusters/postgresql_controller.go:285]: It was needed when we used apiv1, to set twofactordelete on creation. Removed

File: pkg/instaclustr/client.go [hunk 5]
```diff
@@ -145,6 +146,10 @@ func (c *Client) UpdateOpenSearch(id string, o models.OpenSearchInstAPIUpdateReq
 		return err
 	}
 
+	if strings.Contains(string(body), ClusterIsNotInStateToResize) {
+		return ClusterIsNotReadyToResize
+	}
```
Comment (ribaraka) [inline pkg/instaclustr/client.go:151]: I think it's better to use status Conflict

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: pkg/models/operator.go [hunk 6]
```diff
@@ -147,6 +147,7 @@ const (
 const Requeue60 = time.Second * 60
 
 var (
-	ReconcileRequeue = reconcile.Result{RequeueAfter: Requeue60}
-	ExitReconcile    = reconcile.Result{}
+	ReconcileRequeue    = reconcile.Result{RequeueAfter: Requeue60}
+	ReconcileRequeueNow = reconcile.Result{Requeue: true}
```
[the remark this answers is not in the document]
Comment (RostislavPorohnya) [reply pkg/models/operator.go:151]: Ok

File: pkg/instaclustr/client.go (hunk 5 above)
Comment (RostislavPorohnya) [reply pkg/instaclustr/client.go:151]: Status Conflict can be error other from "not ready to resize"

File: pkg/instaclustr/client.go [hunk 7]
```diff
@@ -1750,6 +1784,10 @@ func (c *Client) UpdatePostgreSQLDataCentres(id string, dataCentres []*models.PG
...
```
[the remark this answers is not in the document]
Comment (RostislavPorohnya) [reply pkg/instaclustr/client.go:1789]: I would check specifically a "not ready to resize" error

[the remark this answers is not in the document]
Comment (ribaraka) [reply pkg/instaclustr/client.go:1789]: Please don't use it. Just change the name of the current "error update" annotation.

File: pkg/instaclustr/errors.go [hunk 8]
```diff
@@ -19,9 +19,11 @@ package instaclustr
...
 var (
-	StatusPreconditionFailed  = errors.New("412 - status precondition failed")
 	ClusterNotRunning         = errors.New("сluster is not running")
 	NotFound                  = errors.New("not found")
-	HasActiveResizeOperation  = errors.New("cluster has active resize operation")
 	ClusterIsNotReadyToResize = errors.New("cluster is not ready to resize yet")
 )
+
+var (
+	ClusterIsNotInStateToResize = "Your cluster is not in a state where it can be resized at the moment"
```
Comment (ribaraka) [inline pkg/instaclustr/errors.go:28]: remove it please

**Review by ribaraka — CHANGES_REQUESTED** [review]

File: pkg/instaclustr/errors.go (hunk 8 above)
Comment (RostislavPorohnya) [reply pkg/instaclustr/errors.go:28]: Removed

**Review by ribaraka — APPROVED** [review]

**Review by testisnullus — APPROVED** [review]

**Review by taaraora — APPROVED** [review]
